optimum/habana/transformers/trainer.py

```diff
@@ -622,6 +622,8 @@ def hpu_deepspeed_checkpointing(function, *checkpoint_args):
                 import transformers.models.t5.modeling_t5 as modeling_t5
 
                 modeling_t5.checkpoint = torch.utils.checkpoint.checkpoint
+        else:
+            self.model.gradient_checkpointing_disable()
```

Comment:
```suggestion
            # Hack because `RegressionModel` in test_trainer.py doesn't have `gradient_checkpointing_disable`
            if hasattr(self.model, "gradient_checkpointing_disable"):
                self.model.gradient_checkpointing_disable()
```

Reply:
Updated as per suggestion, thanks.